File: BotsSDK/korebot/src/main/java/kore/botssdk/adapter/ComposebarAttachmentAdapter.java
package kore.botssdk.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;
import com.bumptech.glide.request.target.DrawableImageViewTarget;

import java.util.ArrayList;
import java.util.HashMap;

import kore.botssdk.R;
import kore.botssdk.listener.AttachmentListner;
import kore.botssdk.utils.StringUtils;
import kore.botssdk.view.viewUtils.FileUtils;

public class ComposebarAttachmentAdapter extends RecyclerView.Adapter<ComposebarAttachmentAdapter.ImageAttachView> {

    final Context context;
    final AttachmentListner attachmentListner;

    public ComposebarAttachmentAdapter(Context context, AttachmentListner attachmentListner) {
        this.context = context;
        this.attachmentListner = attachmentListner;
    }

    @NonNull
    @Override
    public ImageAttachView onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.attachment_view_composebar, parent, false);
        return new ImageAttachView(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ImageAttachView holder, int position) {
        String fileExtn = dataList.get(position).get("fileExtn");
        if (FileUtils.ImageTypes().contains(fileExtn)) {
            Glide.with(context).load(dataList.get(position).get("localFilePath")).apply(new RequestOptions().diskCacheStrategy(DiskCacheStrategy.NONE)).into(new DrawableImageViewTarget(holder.attachView));
        } else {
            holder.attachView.setImageResource(FileUtils.getDrawableByExt(!StringUtils.isNullOrEmptyWithTrim(fileExtn) ? fileExtn.toLowerCase() : ""));
        }
        holder.closeIcon.setOnClickListener(view -> {
            if (dataList.isEmpty()) return;
            dataList.remove(holder.getBindingAdapterPosition());
            notifyItemRangeInserted(0, dataList.size() - 1);
            attachmentListner.onRemoveAttachment();
        });
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }

    ArrayList<HashMap<String, String>> dataList = new ArrayList<>();

    public void addAttachment(HashMap<String, String> attachmentKey) {
        dataList.add(attachmentKey);
        notifyDataSetChanged();
    }

    public void clearAll() {
        dataList = new ArrayList<>();
        notifyDataSetChanged();
    }

    public ArrayList<HashMap<String, String>> getData() {
        return dataList;
    }


    static class ImageAttachView extends RecyclerView.ViewHolder {
        final View closeIcon;
        final ImageView attachView;

        public ImageAttachView(@NonNull View itemView) {
            super(itemView);
            closeIcon = itemView.findViewById(R.id.close_icon);
            attachView = itemView.findViewById(R.id.attach_view);
        }
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/adapters/ComposeBarAttachmentAdapter.kt
package com.kore.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.DrawableImageViewTarget
import com.kore.common.constants.MediaConstants.Companion.EXTRA_LOCAL_FILE_PATH
import com.kore.common.constants.MediaConstants.Companion.EXTRA_MEDIA_TYPE
import com.kore.common.utils.FileUtils
import com.kore.listeners.AttachmentListener
import com.kore.ui.R

class ComposeBarAttachmentAdapter(
    val context: Context,
    private val attachmentListener: AttachmentListener,
    private val dataList: List<Map<String, *>>
) : RecyclerView.Adapter<ComposeBarAttachmentAdapter.ImageAttachView>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageAttachView {
        val view: View =
            LayoutInflater.from(context).inflate(R.layout.attachment_view_composebar, parent, false)
        return ImageAttachView(view)
    }

    override fun onBindViewHolder(holder: ImageAttachView, position: Int) {

        when(val fileExtension: String = dataList[position][EXTRA_MEDIA_TYPE] as String)
        {
            "jpg","png","jpeg" -> {
                Glide.with(context).load(dataList[position][EXTRA_LOCAL_FILE_PATH])
                    .apply(RequestOptions().diskCacheStrategy(DiskCacheStrategy.NONE))
                    .into<DrawableImageViewTarget>(DrawableImageViewTarget(holder.attachView))
            }
            else -> {
                Glide.with(context).load(getDrawableByExt(fileExtension))
                    .apply(RequestOptions().diskCacheStrategy(DiskCacheStrategy.NONE))
                    .into<DrawableImageViewTarget>(DrawableImageViewTarget(holder.attachView))
            }
        }

        holder.closeIcon.setOnClickListener {
            notifyItemRangeInserted(0, dataList.size - 1)
            attachmentListener.onRemoveAttachment()
        }
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    fun getData(): List<Map<String, *>> {
        return dataList
    }


    class ImageAttachView(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val closeIcon: View
        val attachView: ImageView

        init {
            closeIcon = itemView.findViewById(R.id.close_icon)
            attachView = itemView.findViewById(R.id.attach_view)
        }
    }

    private fun getDrawableByExt(ext: String): Int {
        return if (FileUtils.videoTypes().contains(ext)) {
            com.kore.botclient.R.drawable.ic_video
        }  else if (FileUtils.spreadTypes().contains(ext)) {
            com.kore.botclient.R.drawable.ic_sheet_old
        } else if (FileUtils.docType().contains(ext)) {
            com.kore.botclient.R.drawable.ic_document_old
        } else if (FileUtils.pdf == ext) {
            com.kore.botclient.R.drawable.ic_pdf
        } else {
            com.kore.botclient.R.drawable.ic_file_general
        }
    }

}